src/tree/mod.rs

```diff
@@ -0,0 +1,13 @@
+//! The tree functionality
+//!
+
+pub mod document;
+pub mod namespace;
+pub mod node;
+pub mod nodetype;
+
+pub use tree::document::Document;
+pub(crate) use tree::document::DocumentRef;
+pub use tree::namespace::Namespace;
+pub use tree::node::Node;
+pub use tree::nodetype::NodeType;
```

Comment:
I'm quite happy that this reexporting technique of `pub use` works so well in Rust. Because of it there is absolutely no breaking changes to the public-facing API, and the restructuring can be entirely under the hood.